File: Algorithms/Sorting_Algorithms/brickSort.py
def brickSort(array, n):
    isSorted = 0
    while isSorted == 0:
        isSorted = 1
        temp = 0
        for i in range(1, n - 1, 2):
            if array[i] > array[i + 1]:
                array[i], array[i + 1] = array[i + 1], array[i]
                isSorted = 0

        for i in range(0, n - 1, 2):
            if array[i] > array[i + 1]:
                array[i], array[i + 1] = array[i + 1], array[i]
                isSorted = 0
    return


arr = [23,45,67,89]
n = len(arr)

brickSort(arr, n);
for i in range(0, n):
    print(arr[i], end=' ')

File: Algorithms/Sorting_Algorithms/BrickSort.js
/* A JavaScript program to perform Brick Sort */

// A utility function to sort the array values
function swap(arr, first_Index, second_Index) {
  var temp = arr[first_Index];
  arr[first_Index] = arr[second_Index];
  arr[second_Index] = temp;
}

// A function to perform the Brick Sort
function brickSort(arr) {
  if (arr.length < 1) return 0;
  var arrayLength = arr.length;
  var isSorted = false;
  while(!isSorted){
  isSorted = true;
  
  // Perform Bubble sort on odd indexed element
  for (let i=1; i<=arrayLength-2; i=i+2) 
    { 
      if (arr[i] > arr[i+1]) 
        { 
          swap(arr,i,i+1);
          isSorted = false; 
        } 
     }
     
  // Perform Bubble sort on even indexed element
  for (let i=0; i<=arrayLength-2; i=i+2) 
    { 
       if (arr[i] > arr[i+1]) 
         { 
           swap(arr,i,i+1);         
           isSorted = false; 
         } 
     }
   }
   return arr;
}
console.log(brickSort([3, 0, 2, 5, -1, 4, 1]));
